daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/TraceLog.scala

```diff
@@ -4,28 +4,30 @@
 package com.daml.lf.speedy
 
 import com.daml.lf.data.Ref.Location
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
 import org.apache.commons.text.StringEscapeUtils
-import org.slf4j.Logger
 
 private[lf] trait TraceLog {
-  def add(message: String, optLocation: Option[Location]): Unit
+  def add(message: String, optLocation: Option[Location])(implicit
+      loggingContext: LoggingContext
+  ): Unit
   def iterator: Iterator[(String, Option[Location])]
 }
 
-private[lf] final case class RingBufferTraceLog(logger: Logger, capacity: Int) extends TraceLog {
+private[lf] final class RingBufferTraceLog(logger: ContextualizedLogger, capacity: Int)
+    extends TraceLog {
 
   private val buffer = Array.ofDim[(String, Option[Location])](capacity)
   private var pos: Int = 0
   private var size: Int = 0
 
-  def add(message: String, optLocation: Option[Location]): Unit = {
-    if (logger.isDebugEnabled) {
-      logger.debug(
-        "{}: {}",
-        StringEscapeUtils.escapeJava(Pretty.prettyLoc(optLocation).renderWideStream.mkString),
-        StringEscapeUtils.escapeJava(message),
-      )
-    }
+  def add(message: String, optLocation: Option[Location])(implicit
+      loggingContext: LoggingContext
+  ): Unit = {
+    logger.debug(
+      StringEscapeUtils.escapeJava(Pretty.prettyLoc(optLocation).renderWideStream.mkString) + ": " +
+        StringEscapeUtils.escapeJava(message)
```

Comment:
Is this the only place from which you are logging, specifically using the `ContextualizedLogger`?

Reply:
In this PR, there are 2 places. Here and in `WarningLog`. 
I would like also to log Speedy internal errors with a normal `ContextualizedLogger`.